modules/js/perf/perf_imgproc/perf_cvtcolor.js

```diff
@@ -4,20 +4,36 @@ if (isNodeJs) {
   var Benchmark = require('benchmark');
   var cv = require('../../opencv');
   var HelpFunc = require('../perf_helpfunc');
-  var Base = require('../base');
 } else {
   var paramsElement = document.getElementById('params');
   var runButton = document.getElementById('runButton');
   var logElement = document.getElementById('log');
 }
 
-cv.onRuntimeInitialized = () => {
+var cvSize
+
+async function main() {
+  if (cv instanceof Promise) {
+    cv = await cv
+  }
+
+  cvSize = {
```

Comment:
Put cvSize definition in every files are duplicated. I understand the calling to cv.Size is not valid before cv promise is resolved. Could you please create a function inside base.js (e.g. createCvSize) that returns the cvSize? You then can call that function after await.